secure_fw/partitions/internal_trusted_storage/tfm_its_req_mngr.c

```diff
@@ -130,6 +154,27 @@ static psa_status_t tfm_its_get_req(const psa_msg_t *msg)
     out_size = msg->out_size[0];
     first_get = true;
 
+#ifdef TFM_ITS_ENCRYPTED
+/* The buffer check is only need if we don't use the input buffer directly */
+#if PSA_FRAMEWORK_HAS_MM_IOVEC != 1
+    /* The PS may also encrypt the data so we don't encrypt it's content. Thus
+     * the limitations regarding the ITS encryption are not present.
+     */
+#ifdef TFM_PARTITION_PROTECTED_STORAGE
+    if (asset_info.client_id != TFM_SP_PS) {
+#else
+    {
+#endif /* TFM_PARTITION_PROTECTED_STORAGE */
+        /* When encryption is enabled the whole file needs to fit in the
+         * global buffer.
+         */
+        if(out_size > sizeof(asset_data)){
+            return PSA_ERROR_BUFFER_TOO_SMALL;
+        }
+    }
+#endif
+#endif
+
```

Comment:
```suggestion
   status = buffer_size_check(asset_info.client_id, out_size);
   if(status != PSA_SUCCESS) {
     return status;
   }
```

Let's re-use this code and minimize the amount of disturbance to the non-encrypted ITS.

Reply:
good point, gonna put this in a sperate function